pkg/skaffold/deploy/helm.go

```diff
@@ -308,18 +309,25 @@ func (h *HelmDeployer) deployRelease(ctx context.Context, out io.Writer, r lates
 		return nil, fmt.Errorf("release args: %w", err)
 	}
 
-	iErr := h.exec(ctx, out, r.UseHelmSecrets, args...)
-
+	err = h.exec(ctx, out, r.UseHelmSecrets, args...)
 	var b bytes.Buffer
-
-	// Be accepting of failure
-	if err := h.exec(ctx, &b, false, getArgs(helmVersion, releaseName, opts.namespace)...); err != nil {
-		logrus.Warnf(err.Error())
-		return nil, nil
-	}
-
 	artifacts := parseReleaseInfo(opts.namespace, bufio.NewReader(&b))
-	return artifacts, iErr
+	if err != nil {
+		return artifacts, fmt.Errorf("install: %w", err)
+	}
+
+	args = getArgs(helmVersion, releaseName, opts.namespace)
+	// At one point in ancient history (#644), it was determined that "helm get"
+	// immediately after install would sometimes be unreliable. We carry that
+	// tradition forward today by retrying once rather than ignoring the result.
+	if err := h.exec(ctx, &b, false, args...); err != nil {
+		logrus.Warnf("helm get failed - will retry: %v", err)
+		time.Sleep(5 * time.Second)
```

Comment:
Do we use `--wait` option on `helm install`? 

The only worry i have with this constant `sleep` is
would 5 seconds be enough (for huge helm app) or too much (for small one)?

Reply:
Good point. Apparently the deployer passes in `--wait` if the user specifies it, but it's optional. What do you think about replacing the retry with:

* If `r.Wait == true`, try once and return the error code
* If `r.Wait == false`, try once, warn the user if get fails, but mention that they need to set `Wait` in their config to be fatal.

Reply:
+1 to this approach.

Reply:
I tried to think about what the user wants most out of this functionality, and came up with the following compromise:

- if r.Wait == true, retry until the condition is met.
- if r.Wait == false, retry for up to 2 seconds, and issue a warning. It's worth noting that "get" succeeding does not mean the application is fully deployed, only that helm is aware of the deployment.

PTAL.